### ISSUE TITLE
✨ Hold down after delay instead of tapping.

### ISSUE THREAD

**Issue by JaceThings** [MrKai77/Loop#111]
### Feature Request Description

I think it would be cool to allow holding down a key to cycle between sizes in a cycle. After a certain amount of time holding it, it would start cycling through the options of that particular key. It just feels a lot more intuitive than tapping repeatedly.

### Final Checks

- [X] My issue title is descriptive
- [X] This is a single feature request (multiple feature requests should be reported individually)

**Comment by MrKai77** [comment]
Interesting, approach! Will look into this!

**Comment by MrKai77** [comment]
So would Loop cycle through each cycle action after a user-set delay? Or did you have a type of delay in mind? (maybe there's something similar built-in to MacOS?)

**Comment by JaceThings** [comment]
> So would Loop cycle through each cycle action after a user-set delay?

Pretty much! I thought about this over the weekend, and I thought the most "helpful" thing would be two timers:

1. Delay to hold to activate the "auto cycle."
2. "Decision Delay", which would be the amount of time the cycle appears on the screen for.

So this would be the Scenario.

Auto Cycle Delay: 5 Seconds
Decision Delay: 3 seconds

1. I hold down my activator keys. 
2. The first item (Action 0 for this example) in the cycle appears on the screen. 
3. I wait for 5 seconds. 
4. Action 1 then appears. 
5. I now wait for 3 seconds. 
6. Action 2 appears. 
7. I now wait for 3 seconds
8. ...

The "Decision Delay" is an extra feature I added to allow the user to avoid an initial delay when accessing action 0. If there was only one delay, pressing and holding the key would cycle through the actions too quickly when holding the action key down and wanting to activate the 0th action.

> (maybe there's something similar built-in to MacOS?)

Closest thing i can think of is when you hold down a key in macOS and you have "repeat letter" enabled.

**Comment by MrKai77** [comment]
That makes a lot of sense, thanks!
I was also thinking that it would be hard to access the first action if there was only one delay, but having two would be much easier to use!
Anyways, this does seem like a more complicated feature compared to the other ones currently in Loop, so I think that it might be better to implement this after reorganizing/redesigning the settings window. (as this would probably be a more of a power-user feature)

**Comment by JaceThings** [comment]
Yep, agreed. I would possibly add it to its own tab/section, depending on how it's structured. I will reply to the other settings design ticket soon; finishing work!

**Comment by MrKai77** [comment]
Awesome, thank you so much!
I was thinking of structuring it so that there's a tab dedicated to Loop's behaviour when resizing windows, meaning it would contain this issue, as well as these other settings:
<img width="419" alt="image" src="https://github.com/MrKai77/Loop/assets/68963405/0cf7baaf-1d98-4b96-b884-7ae0b296545c">

Also: I think that we should move the settings window redesign into its own issue, since it's a little unrelated to this issue or #183 :)

**Comment by MrKai77** [comment]
Randomly thought of this but what do you think if you could click while Loop is open to go to the next cycling action?

**Comment by JaceThings** [comment]
I don't think I follow? 

Can't one cycle be "active" at one time?

Isn't that you described just what loop does normally?

**Comment by MrKai77** [comment]
Ohhh sorry I should explain this more specifically haha
This would only work when using a mouse (so it'd probably not suit your use case), but I think it's a cool idea nonetheless :)

So basically, this is how it would work:
1. Trigger Loop with <kbd>Trigger Key</kbd>
2. Move cursor to a side (eg. Right) to select a window action (Right Cycle)
3. Click using mouse/trackpad to trigger the next cycling action (so Right Half would become Right Two Thirds and then Right Third and so on...)
4. Keep clicking until the desired window action is selected
5. Release <kbd>Trigger Key</kbd> to activate that action within the cycle!

**Comment by JaceThings** [comment]
I like this, would make a lot of sense actually.

**Comment by MrKai77** [comment]
Closing, as I no longer think adding a delay before advancing to the next cycle item makes sense, as it could easily result in unintentional selections, or could end up feeling slow otherwise. The click-based solution has also been implemented, and it feels like a more sensible approach in my opinion.
